ext/mixed/js/display.js

```diff
@@ -354,15 +367,29 @@ class Display {
         }
     }
 
+    onProfileChanged(profileIndex=null) {
+        if (profileIndex !== null) {
+            this.profileSwitcher.setIndex(profileIndex);
+        }
+        this.renderProfileSelect();
+        const options = this.options;
+        this.updateDocumentOptions(options);
+        this.updateTheme(options.general.popupTheme);
+        this.setCustomCss(options.general.customPopupCss);
+    }
+
     getOptionsContext() {
-        throw new Error('Override me');
+        return {
+            index: this.profileSwitcher.globalProfileIndex
```

Comment:
This could be safer if instead of the index a unique ID was used. It could be either autoincrement from the DB or a randomly generated number. Not sure if same applies to backend/options because the index comes from there.